app/src/main/java/com/eveningoutpost/dexdrip/Services/MissedReadingService.java

```diff
@@ -42,20 +43,26 @@ protected void onHandleIntent(Intent intent) {
         bg_missed_minutes =  Integer.parseInt(prefs.getString("bg_missed_minutes", "30"));
         otherAlertSnooze =  Integer.parseInt(prefs.getString("other_alerts_snooze", "20"));
 
-        if (bg_missed_alerts && BgReading.getTimeSinceLastReading() > (bg_missed_minutes * 1000 * 60)) {
+        if (bg_missed_alerts && BgReading.getTimeSinceLastReading() > (bg_missed_minutes * 1000 * 60)
+                &&
+                prefs.getLong("alerts_disabled_until", 0) < new Date().getTime()) {
```

Comment:
I would replace the > in >= 
and the < in <=.
The reason for that is assume that we ask for a re-check in 30 minutes, and indeed we check after 30 minutes exactly, the alarm will not fire...